mitodl/mitxpro
# Fixed race conditions around user creation and repair scripts

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #618

#### What's this PR do?
- Fixes a race condition that was present if the user record got created and the task executed before the transaction committed
- Adds a management command to repair existing users' courseware records
- Updates the existing command to repair edx enrollments to walk all of the ones that statisfy `edx_enrolled=False`, or the old behavior of passing a username

#### How should this be manually tested?
- You should be able to sign up and get the expected courseware records created without encountering an error.
- You should be able to run `./manage.py repair_missing_courseware_records` to repair missing `OpenEdxApiAuth` and/or `CoursewareUser` records (note that deleting `CoursewareUser` and re-attempting creation will fail because the user exists in openedx)
- You should be able to run `./manage.py retry_edx_enrollment [USERNAME [USERNAME ...]]` to retry enrollments for all or a specific set of users.
